File: leetcode/leetcode718.py
class Solution:
    def findLength(self, A: List[int], B: List[int]) -> int:
        length_a = len(A)
        length_b = len(B)
        dp = [[0]*(length_a+1) for _ in range(length_b)]
        for i in range(length_a + 1):
            if A[i] == B[i]:
                dp[i][i] = 1
        for i in range(length_a + 1):
            if A[0] == B[i]:
                dp[0][i] = 1

        for i in range(1,length_b+1):
            for j in range(1, length_a + 1):
                if A[j] == B[i]:
                    dp[i][j] = max(dp[i-1][j], dp[i][j-1]) + 1

        return max(max(x) for x in dp)

File: java/Leetcode718.java
public class Leetcode718 {
    public int findLength(int[] A, int[] B) {
        int[][] dp = new int[A.length+1][B.length+1];
        int ans = 0;
        for (int i = 1; i < A.length + 1; i++) {
            for (int j = 1; j < B.length + 1; j++) {
                if (A[i - 1] == B[j - 1]){
                    dp[i][j] = Math.max(dp[i][j], dp[i-1][j-1] + 1);
                    ans = Math.max(dp[i][j], ans);
                }

            }
        }
        return ans;
    }

    public static void main(String[] args) {

        System.out.println(new Leetcode718().findLength(
                new int[] {1,2,3,2,1},
                new int[] {3,2,1,4,7}
        ));
    }
}
